aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -257,6 +258,13 @@ private void newStream() throws IOException {
 
   @Override
   public void close() throws IOException {
+
+    // A failed s3 close removes state that is required for a successful close.
```

Comment:
nit: wondering if the following comment is a little more clear based on Russel's comment

```
A previously failed close left the S3 output stream in a bad state that is non-recoverable.
Any future close on this stream should fail.
```

Reply:
updated based on suggestion.